aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/LoggingProxyClient.java

```diff
@@ -76,5 +79,19 @@ private <RequestT extends AwsRequest, ResultT> ResultT logAndDelegate(
         return result;
     }
 
-
+    private <RequestT extends AwsRequest, ResultT> ResultT logRequestAndDelegate(
+            final RequestT request,
+            final Function<RequestT, ResultT> requestFunction,
+            final BiFunction<RequestT, Function<RequestT, ResultT>, ResultT> injectCredentials
+    ) {
+        ResultT result = null;
+        try {
+            requestLogger.log(request);
+            result = injectCredentials.apply(request, requestFunction);
+        } catch (Exception e) {
+            requestLogger.logAndThrow(e);
+        }
+        requestLogger.log("[Result log omitted]");
+        return result;
+    }
```

Comment:
Logging the response here and in `AmazonWebServicesClientProxy.logRequestMetadataV2` (See https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/issues/414 for more details) caused our paginators to execute 3x more requests than needed.

Reply:
Does only logging request records RequestId?